api/desecapi/pdns_change_tracker.py

```diff
@@ -0,0 +1,228 @@
+                _pdns_post('/zones', payload)
+
+            additions = self._rr_set_additions.get(domain_name, set())
+            modifications = self._rr_set_modifications.get(domain_name, set())
+            deletions = self._rr_set_deletions.get(domain_name, set())
+
+            assert additions & deletions == set()
+            assert modifications & deletions == set()
+
+            # Due to disjoint guarantees with `deletions`, we have four types of RR sets:
+            # (1) purely added RR sets
+            # (2) purely modified RR sets
+            # (3) added and modified RR sets
+            # (4) purely deleted RR sets
+
+            # We send RR sets to PDNS if one of the following conditions holds:
+            # (a) RR set was added and has at least one RR
+            # (b) RR set was modified
+            # (c) RR set was deleted
+
+            # Conditions (b) and (c) are already covered in the modifications and deletions list,
+            # we filter the additions list to remove newly-added, but empty RR sets
+            additions -= set([
+                (type_, subname) for (type_, subname) in additions
+                if not RR.objects.filter(
+                    rrset__domain__name=domain_name,
+                    rrset__type=type_,
+                    rrset__subname=subname).exists()
+            ])
+
+            data = {
+                'rrsets': [
+                    {
+                        'name': '.'.join(filter(None, [subname, domain_name])) + '.',
+                        'type': type_,
+                        'ttl': self._get_ttl_or_zero(domain_name, type_, subname),
+                        'changetype': 'REPLACE',
+                        'records': [
+                            {'content': rr.content, 'disabled': False}
+                            for rr in RR.objects.filter(
+                                rrset__domain__name=domain_name,
+                                rrset__type=type_,
+                                rrset__subname=subname)
+                        ]
+                    }
+                    for type_, subname in (additions | modifications) - deletions
+                ] + [
+                    {
+                        'name': '.'.join(filter(None, [subname, domain_name])) + '.',
+                        'type': type_,
+                        'changetype': 'REPLACE',
+                        'records': []
+                    }
+                    for type_, subname in deletions
+                ]
+            }
+
+            if data['rrsets']:
+                _pdns_patch('/zones/' + pdns_id, data)
+
+            if data['rrsets'] or domain_name in self._domain_additions:
+                _pdns_put('/zones/%s/notify' % pdns_id)
+
+    @staticmethod
+    def _get_ttl_or_zero(domain_name, type_, subname):
+        result = RRset.objects.filter(domain__name=domain_name, type=type_, subname=subname)
```

Comment:
To avoid constructing the full object, you can do
```python
result = RRset.objects.filter(domain__name=domain_name, type=type_, subname=subname).values_list('ttl', flat=True)
return result[0] if result else 0
```